eng/scripts/docs/tests/Toc-Generation-Tests.ps1

```diff
@@ -0,0 +1,76 @@
+<#
+```
+Invoke-Pester -Output Detailed $PSScriptRoot\tests\Toc-Generation-Tests.ps1
```

Comment:
Lets see if we can also add some tests for the Toc functions as well.

Reply:
`Get-Toc-Children` is worth to have more tests as they have some complicated logic. I will leave the `Get-dotnet-DocsMsTocData` as it is since the logic is pretty straight forward and it is the same structure as other languages.